[Data Streams] Docs: hidden ≠ Elasticsearch system data streams (#280089)
elastic/kibana: src/platform/packages/private/kbn-data-streams/README.md

## System data streams

`@kbn/data-streams` (and Core `registerDataStream`) can create **hidden** data streams (`hidden` defaults to `true`). That is **not** the same as an Elasticsearch **system** data stream.

| Flag | Who sets it | What it means |
|------|-------------|----------------|
| `hidden: true` | Kibana definition / index template | Stream and backing indices are hidden from normal listings |
| `system: true` | Elasticsearch only | Stream is registered as a system data stream (security / restricted-index semantics, product origin, DLM treatment, etc.) |

Kibana **cannot** mark a stream as system. That requires a matching [`SystemDataStreamDescriptor`](https://javadoc.io/doc/org.elasticsearch/elasticsearch/latest/org/elasticsearch/indices/SystemDataStreamDescriptor.html) in the Elasticsearch Kibana plugin (or equivalent ES registration). If you only call `registerDataStream` / `DataStreamClient.initialize` without that ES registration, `GET _data_stream/<name>` typically reports `system: false` even when the stream is hidden and named `.kibana_*` / `.workflows-*`.

### Why this matters

> Privileged Kibana streams that are **not** Elasticsearch system data streams have been readable across spaces (the class of bug behind [security-team#18291](https://github.com/elastic/security-team/issues/18291)). Treat any stream that holds privileged or cross-space-sensitive data as needing ES `system: true` — `hidden` alone is not enough.

Reviewer: ```suggestion
`@kbn/data-streams` are **not** system data streams by default. This means data is readable across spaces by default (leading to the class of bug behind [security-team#18291](https://github.com/elastic/security-team/issues/18291)). Treat any stream that holds privileged or cross-space-sensitive data as needing ES `system: true` — `hidden` alone is not enough.
```
